runtime/tests/checker/interface_test.go

```diff
@@ -1991,3 +1998,117 @@ func TestCheckInvalidInterfaceUseAsTypeSuggestion(t *testing.T) {
 		errs[0].(*sema.InvalidInterfaceTypeError).ExpectedType,
 	)
 }
+
+//MultipleInterfaceDefaultImplementation
+func TestCheckInvalidMultipleInterfaceDefaultImplementation(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+	struct interface InterfaceA {
+		fun test(): Int{
+			return 41
+		}
+	}
+
+	struct interface InterfaceB {
+		fun test(): Int{
+			return 41
+		}
+	}
+	
+	struct Test:  InterfaceA, InterfaceB   {
+
+	}
+
+    `)
+
+	errs := ExpectCheckerErrors(t, err, 1)
+
+	require.IsType(t, &sema.MultipleInterfaceDefaultImplementationsError{}, errs[0])
+
+}
+
+func TestCheckMultipleInterfaceDefaultImplementationWhenOverriden(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+	struct interface InterfaceA {
+		fun test(): Int{
+			return 41
+		}
+	}
+
+	struct interface InterfaceB {
+		fun test(): Int{
+			return 41
+		}
+	}
+	
+	struct Test:  InterfaceA, InterfaceB   {
+		fun test(): Int{
+			return 42
+		}
+	}
+
+
+    `)
+
+	require.NoError(t, err)
+
+}
+
+func TestCheckMultipleInterfaceSingleInterfaceDefaultImplementation(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+	struct interface InterfaceA {
+		fun test(): Int{
+			return 41
+		}
+	}
+
+	struct interface InterfaceB {
+		fun test(): Int
+	}
+	
+	struct Test:  InterfaceA, InterfaceB  {
+
+	}
+
+    `)
+
+	errs := ExpectCheckerErrors(t, err, 1)
+
+	require.IsType(t, &sema.ConformanceError{}, errs[0])
```

Comment:
I do not understand why this should be an error. Why it it not allowed to use the default value from InterfaceA here?

Reply:
Normally if the interface does not have a default implementation, later on, adds one, should be a non-breaking change for the contract that implements that interface. But if we allow this case, then we lose this guarantee.

Reply:
This make sense as long as you only allow a single interface to have a default implementation.

Reply:
@bjartek sorry I don't follow, can you explain a bit more?

Reply:
I see your point on why it needs to be like this :)

Reply:
Actually at first I was thinking order of conformances could be used, but it can be little confusing to new developers.

Reply:
Agreed that this should be an error, and using order of conformances is not a good idea, as it might not be obvious 👍